specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/SessionPools.json

```diff
@@ -0,0 +1,702 @@
+            },
+            "secrets": {
+              "description": "The secrets of the Session Pool.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Secret"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "dynamicPoolConfiguration": {
+              "description": "The pool configuration if the poolManagementType is dynamic.",
+              "$ref": "#/definitions/DynamicPoolConfiguration"
+            },
+            "customContainerTemplate": {
+              "description": "The custom container configuration if the containerType is node-lts.",
+              "$ref": "#/definitions/CustomContainerTemplate"
+            },
+            "sessionNetworkConfiguration": {
+              "description": "The network configuration of the Sessions in the Session Pool.",
+              "$ref": "#/definitions/SessionNetworkConfiguration"
+            },
+            "poolManagementEndpoint": {
+              "description": "The endpoint to manage the pool.",
+              "type": "string",
+              "readOnly": true
+            },
+            "provisioningState": {
+              "description": "Provisioning state of the Session Pool.",
+              "enum": [
+                "InProgress",
+                "Succeeded",
+                "Failed",
+                "Canceled",
+                "Deleting"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "SessionPoolProvisioningState",
+                "modelAsString": true
+              }
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SessionPoolPatchProperties": {
+      "description": "Container App SessionPool patchable properties.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "SessionPool resource specific patchable properties.",
+          "type": "object",
+          "properties": {
+            "scaleConfiguration": {
+              "description": "The scale configuration of the Session Pool.",
+              "$ref": "#/definitions/ScaleConfiguration"
+            },
+            "secrets": {
+              "description": "The secrets of the Session Pool.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Secret"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "dynamicPoolConfiguration": {
+              "description": "The pool configuration if the poolManagementType is dynamic.",
+              "$ref": "#/definitions/DynamicPoolConfiguration"
+            },
+            "customContainerTemplate": {
+              "description": "The custom container configuration if the containerType is node-lts.",
```

Comment:
>is node-lts. [](http://example.com/codeflow?start=86&length=12)

there are other enums too as in swagger. Can this be generic description ?

Reply:
Only when the containerType is node-lts, users need to specify customContainerTemplate, and only when the poolType is dynamic, users needs to specify dynamicPoolConfiguration.
May I ask what do you mean by generic description? Thanks

Reply:
In future if we need support for another container type under custom container then it will be confusing to users as description gives the impression that it's only applicable for `node-lts`